src/core/document.js

```diff
@@ -656,7 +656,32 @@ class Page {
         }
 
         return Promise.all(annotationPromises).then(function (annotations) {
-          return annotations.filter(annotation => !!annotation);
+          if (annotations.length === 0) {
```

Comment:
I think it can be simplified in using sort function.
I just tried:
```js
["a",1, 2, "d", "e", 3, "aa"].sort((a, b) => typeof a === typeof b ? 0 : (typeof a === "string" ? -1 : +1))
```
and it works.

Reply:
To be honest, I'm not finding such a code-snippet particularly easy to read/understand at a glance.
Also, wouldn't you still need to `filter` out any empty entries first? If so, that'd effectively require *two* loops through the Array rather than just the one.